src/auth/AuthClientHandler.h

```diff
@@ -31,18 +33,19 @@ class AuthClientHandler {
   uint32_t want;
   uint32_t have;
   uint32_t need;
-  RWLock lock;
+  mutable SharedMutexT<lock_policy> lock;
```

Comment:
i wonder whether `AuthClientHandler` itself even needs explicit locking - doesn't `MonClient` already hold its `monc_lock` over any calls to `AuthClientHandler`? are there any users outside of `MonClient` that i'm missing?

Reply:
agreed! the `RWLock` in `AuthClientHandler` was introduced in #1636. back to then, we were accessing `MonClient::auth` directly, see https://github.com/ceph/ceph/commit/a2eb6ae3fb57b09efdd4d7baac6871ca8dd8e79f#diff-fa6c2eba8356ae1442d1bf749beacfdfL6209 . and in #11128, we added a helper method offering accessing to `MonClient::auth`. this method guards the access to `auth` using `monc_lock` .

as all accesses to `MonClient::auth` are now protected by `monc_lock`, i will drop the `RWLock` in `AuthClientHandler` and its children classes in a follow-up PR.

probably we could differentiate the read/write access with RWLock. as we only update the auth's internal status/keyring when handling `MAuthReply`. most of them, we just call `auth->build_authorizer(service_id)`. but let's leave it to yet another follow-up PR.